mineral-ui/mineral-ui
# [Box] [Flex] [StartEnd] Add Box, Flex, FlexItem, and StartEnd components

### Description
- Add ability for nav items to redirect to another
- Append "[Dev-only]" to appropriate example titles
- Add breakpoint variables to mineralTheme
- Update snapshots for components using createThemedComponent
- Add getResponsiveStyles helper to styles
- Add unit tests for getResponsiveStyles
- Add Box, Flex, FlexItem, and StartEnd components
- Add site demos and examples for those four components
- Update Component Status page
- Add unit tests and snapshots for those four components

### Motivation and context
Closes #350 

### Screenshots, videos, or demo, if appropriate


https://350-layout--mineral-ui.netlify.com/components/layout/

### How to test


1. `cd mineral-ui && npm start`
1. Navigate to Box/Flex/FlexItem/StartEnd
1. Ensure all examples render correctly
1. Check all demo content for correctness, helpfulness, clarity, and typos
1. Spin up a sandbox and try to build some layouts :)

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist




### How does this PR make you feel?

![Abstract](https://media.giphy.com/media/3o7bu79VY9ruoYHPs4/giphy.gif)
